### ISSUE TITLE
Why the view doesn't get updated after the loading?

### ISSUE THREAD

**Issue by Taulant1992** [clientIO/joint#1235]
Hi,
while creating a simple graph with links, I change the colors of the ports using the portProp of the joint js.

Below there is the demonstration:
![png1](https://user-images.githubusercontent.com/32356248/71415093-6c0e2580-265a-11ea-875e-6e8cc7deb381.PNG)

Then I save this graph using the getJSON method supported from joint js.
After that, I want to load the graph but there is the problem of the colors because they get back to the default ones.
Please notice that the colors are assigned by default and while making the creation of the complete graph (linking the blocks with each other), the colors change.
The loading of the graph is done from the method fromJSON supported by joint js,

Below there is the demonstration after loading:
![png2](https://user-images.githubusercontent.com/32356248/71415366-90b6cd00-265b-11ea-8ade-3f5cdf7b7b28.PNG)

As far as I see, the colors in the groups take precedence of those in the items.
Image below:
![png3](https://user-images.githubusercontent.com/32356248/71415469-2a7e7a00-265c-11ea-98c2-5220bf6f8a68.PNG)

The problem was revealed after updating joint js into the last version.
Thanks in advance for any kind of help :)

**Comment by kumilingus** [comment]
It's probably because you were referring to an element once as `.port-body` and once a `circle`. It's not clear in the current implementation, which one would be applied first. Try to use only `.port-label` selector. `el.portProp(portId, 'attrs/.port-label/fill', '#00FF00')`.

**Comment by Taulant1992** [comment]
Hi @kumilingus.
I tried also that but it didn't work, actually i use as model the  joint.shapes.devs.Model.
After some research, I realized that this model is not supported anymore, and that's why it causes this problem. Is there any way that I can continue with the devs.Model or to use another one?

**Comment by kumilingus** [comment]
I see, the color is lost even if you add another port to the model. You can override (or extend the class) `createPortItem()` method. It's important to keep all the values you wish to persist in `inPorts`/`outPorts` attributes.

before: `inPorts: ['in1', 'in2']`
after: `inPorts: [{ id: 'in1', color: 'red' }, { id: 'in2' }]`
```
joint.shapes.devs.Model.prototype.createPortItem = function(group, port) {
    return {
        id: port.id,
        group: group,
        attrs: {
            '.port-label': {
                text: port.text || port.id
            },
            '.port-body': {
                fill: port.color
            }
        }
    };
};

element.addInPort({ id: 'in1', text: 'My Port', color: 'red' });
```

**Comment by Taulant1992** [comment]
Hi @kumilingus.
Based on you suggestion i tried: 
- In joint js file I tried to override the createPortItem method as follow,
```
createPortItem: function(group, port) {

            return {
                id: port.id,
                group: group,
                attrs: {
                    '.port-label': {
                        text: port.text || port.id
                    },
                    '.port-body': {
                        fill: port.color
                    }
                }
            };
        },
```
Then, in my files, first I display the ports as default
`outPorts = [{id: 'next', color: 'red'}];`
```
cell.addOutPort({
    id: 'next', 
    text: 'next', 
    color: 'red',
    r: 6,
    type: 'output',
    magnet: 'active',
    stroke: '#000'
});
```
```
inPorts = [{id: 'in', color: '#FFFF00'}];
cell.addInPort({
  id: 'in', 
  text: 'in', 
  color: '#FFFF00',
  r: 6,
  type: 'input',
  magnet: 'passive',
  stroke: '#000'
});
```
                
Then I make the connection between cells calling the function portPort,
`cell.portProp(portName, 'attrs/.port-body/fill', color);`
The ports get changed, but when I load them, they keep their default colors, although in the items the color is assigned to green.

**Comment by Taulant1992** [comment]
After many efforts working with my friend, we made a work-around solution to fix the problem. We had to debug the joint js library and go directly to the _updatePortGroup method and call the joint.util.deepSupplement inside the for loop giving as parameters the metrics and the item(port item).
Also when reaching to the point for calling the portProp method, we replaced that line with those below:
```
var view = paper.findViewByModel(cell.id);
view.model.portProp(portName, 'attrs/.port-body/fill', color);
view._updatePorts();
```
But, in order to get the wanted behavior, I will skip the call of the graph.fromJSON() and add the cells one by one with the method adddCell of the library. 
Hope this will give you a clear idea on how to cover this case in future release.

**Comment by kumilingus** [comment]
Well, that's not really something I would recommend. There are 2 options here.
1. Don't use `devs` shapes. Use Element Port API with e.g. `standard.Rectangle` and you won't have these problems.
2. Use `devs` shapes, but don't use `portProp()`. In order to modify a port, you need to change `inPorts`/`outPorts` attribute. For instance, if you want to change the color of a single input port, do this:
```
var portId = 'xy';
var inPorts = joint.util.cloneDeep(el.get('inPorts'));
var port = inPorts.find(port => port.id === portId);
port.color = 'blue';
el.set('inPorts', inPorts);
```

**Comment by kumilingus** [comment]
I am closing the issue now as `won't fix`. Use `devs` shapes only if you're not planning to change individual port's attributes. Always prefer to use any other shape together with Element Port API (https://resources.jointjs.com/docs/jointjs/v3.1/joint.html#dia.Element.ports). `devs` shapes are now kept for backwards compatibility reasons only.

**Comment by Taulant1992** [comment]
Hi @kumilingus sorry for responding so late, I was on holidays and hadn't the possibility to communicate.
I will give it a try to your suggestion.
Thank you and have a nice day :)
